### PR TITLE
Feat: only print to stdout

### PR DESCRIPTION
**Pull request by iyear**
### Summary
📄🎨📚

Enhance `vela def` command with CUE schema output options and example. Refactor output logic and update help message.

> _The `vela def` command got a boost_
> _Now it can output CUE schema to use_
> _You can send it to STDOUT_
> _Or a file, how about that?_
> _It also has refactored output and a new example to peruse_

### Walkthrough
* Import `io` package to use `io.Writer` interface for writing CUE schema ([link](https://github.com/kubevela/kubevela/pull/5958/files?diff=unified&w=0#diff-739f4f64cd699090ae6f9705af491dc7e42decf8d0340ef88109bc2333f32bdeR24))
* Add example to command help message to show how to output CUE schema to STDOUT ([link](https://github.com/kubevela/kubevela/pull/5958/files?diff=unified&w=0#diff-739f4f64cd699090ae6f9705af491dc7e42decf8d0340ef88109bc2333f32bdeL1171-R1174))
* Refactor output file creation and writing logic to use a `writer` variable that can be either `os.Stdout` or a file handle ([link](https://github.com/kubevela/kubevela/pull/5958/files?diff=unified&w=0#diff-739f4f64cd699090ae6f9705af491dc7e42decf8d0340ef88109bc2333f32bdeL1184-R1202))
* Change `Writer` field of `providergen.Options` struct to use `writer` variable instead of `f` variable ([link](https://github.com/kubevela/kubevela/pull/5958/files?diff=unified&w=0#diff-739f4f64cd699090ae6f9705af491dc7e42decf8d0340ef88109bc2333f32bdeL1197-R1208))

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: references/cli/def.go [hunk 1]
```diff
@@ -1168,7 +1169,9 @@ func NewDefinitionGenCUECommand(_ common.Args) *cobra.Command {
 			"# Generate CUE schema for provider type with custom types\n" +
 			"> vela def gen-cue -t provider --types *k8s.io/apimachinery/pkg/apis/meta/v1/unstructured.Unstructured=ellipsis /path/to/myprovider.go\n" +
 			"# Generate CUE schema for provider type with custom output path\n" +
-			"> vela def gen-cue -t provider -o /path/to/myprovider.cue /path/to/myprovider.go\n",
+			"> vela def gen-cue -t provider -o /path/to/myprovider.cue /path/to/myprovider.go\n" +
+			"# Output CUE schema to STDOUT\n" +
+			"> vela def gen-cue -t provider -o - /path/to/myprovider.go",
```
Comment (FogDong) [inline references/cli/def.go:1174]: Do we need this `-o` flag? Maybe we can just output the result to stdout, if the user want to write to a file, he can use `vela def gen-cue -t provider > mycue.cue`

Comment (FogDong) [reply references/cli/def.go:1174]: WDYT? @Somefive

Comment (iyear) [reply references/cli/def.go:1174]: hmm, do we need to consider Windows users?

Comment (iyear) [reply references/cli/def.go:1174]: well, my bad, `>` is also supported in powershell

Comment (iyear) [reply references/cli/def.go:1174]: Do we need this: `Output CUE file path, if not specified, the CUE file will be generated in the same directory`? if not, only stdout +1

Comment (iyear) [reply references/cli/def.go]: Updated

**Comment by Somefive** [comment]
Maybe "go mod tidy"? A CI failed.

**Review by Somefive — APPROVED** [review]

**Comment by iyear** [comment]
my bad. let me fix ci error later

**Review by FogDong — APPROVED** [review]

**Pull request merged**